### ISSUE TITLE
`suggest#fix` shouldn't be optional

### ISSUE THREAD

**Issue by rafaelss95** [typescript-eslint/typescript-eslint#3948]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
Currently, the `fix` within `suggest` is optional, however it shouldn't. Even if we forget to use it, [ESLint give us this error](https://github.com/eslint/eslint/blob/58840ac844a61c72eabb603ecfb761812b82a7ed/lib/linter/report-translator.js#L306).

```TS
context.report({
  node,
  messageId,
  suggest: [{ messageId: messageIdSuggest }] // missing `fix` and no error
})
```

**Expected Result**
```ts
context.report({
  node,
  messageId,
  suggest: [{ messageId: messageIdSuggest }] // error - missing `fix`
})
```

**Actual Result**
No errors being shown.

**Additional Info**
Maybe a solution would be:

```diff
type ReportSuggestionArray<TMessageIds extends string> =
- ReportDescriptorBase<TMessageIds>[];
+  (ReportDescriptorBase<TMessageIds> & {
+    readonly fix: ReportFixFunction | null;
+  })[];
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `4.3.2` |
| `TypeScript`                            | `4.4.3` |
| `node`                                  | `14.16.1` |
